src/shogun/base/SGObject.h

```diff
@@ -476,6 +476,47 @@ class CSGObject
 			demangled_type<T>().c_str());
 	}
 
+	/** Typed array getter for an object class parameter of a Shogun base class
+	* type, identified by a name and an index
+	*
+	* @param name name of the parameter array
+	* @param index index of the element in the array
+	*/
+	template <class T,
+			  class X = typename std::enable_if<is_sg_base<T>::value>::type>
+	T* get(const std::string& name, index_t index)
+	{
+		Tag<CDynamicObjectArray*> tag_array_sg(name);
+		if (has(tag_array_sg))
+		{
+			auto array = get(tag_array_sg);
+			if (!array)
+			{
+				SG_ERROR(
+					"Cannot get object  from %s::%s[%d], array "
+					"not instantiated.\n",
+					get_name(), name.c_str(), index);
+			}
+
+			return (T*) get(array, index);
+		}
+
+		Tag<std::vector<T*>> tag_vector(name);
+		if (has(tag_vector))
+		{
+			// TODO this needs test once we have std::vector parameters
+			SG_NOTIMPLEMENTED
+			auto array = get(tag_vector);
+			return array.element_at(index);
```

Comment:
If this is an `std::vector`, then it won't have the `element_at` method, just `at` (if I understand correctly what it is happening here).

Reply:
yep! copy paste error (didnt compile yet)